compendium/ConfigurationAdmin/test/TestConfigAdmin.cpp

```diff
@@ -843,3 +849,100 @@ TEST_F(ConfigAdminTests, testMultipleManagedFactoriesInBundleGetUpdate)
                             1);
                 });
 }
+// <summary>
+/// Used by ConfigAdminTests.testManagedServiceRemoveConfigurationsDeadlock
+/// </summary>
+namespace cppmicroservices {
+namespace test {
+class TestManagedServiceInterface2
+  : public cppmicroservices::service::cm::ManagedService
+{
+public:
+  virtual ~TestManagedServiceInterface2() noexcept = default;
+  void Updated(const cppmicroservices::AnyMap&) override = 0; 
+};
+
+class TestManagedServiceImpl : public TestManagedServiceInterface2
+{
+public:
+  TestManagedServiceImpl(cppmicroservices::Framework& framework)
+    : m_framework(framework)
+   {}
+  virtual ~TestManagedServiceImpl() noexcept = default;
+
+  void Updated(const cppmicroservices::AnyMap& props) override
+  { 
+    // The Updated method is called for both Updated and Removed operations. 
+    // For the Removed operations, the properties are empty.
+    if (props.empty()) {
+      auto ctx = m_framework.GetBundleContext();
+      auto bundles = ctx.GetBundles();
+      for (auto& b : bundles) {
+        b.Stop();
+      }
+    }
+  }
+
+private:
+  cppmicroservices::Framework m_framework;
+};
+}
+}
+/*
+* testManagedServiceRemoveConfigurationsDeadlock
+* 
+* This test was added in response to a deadlock bug. 
+* The Use Case is as follows:
+*     A configuration object is defined in the manifest.json file. (In
+*        this case it is defined in TestBundleManagedServiceDeadlock)
+*     The User's main thread stopped the bundle which causes 
+*         the ConfigurationAdminImpl RemoveConfigurations method 
+*         to remove the configuration object from the ConfigurationAdmin
+*         repository and to send a Removed notification to the service
+*         instance. RemoveConfigurations would then execute a WaitForAllAsync
+*         to wait for all asynchronous threads to complete including the 
+*         asynchronous thread that was launched as part of the Removed 
+*         notification. 
+*      The user's Updated method also tried to stop the bundle. This means
+*          that it had to wait for the RemoveConfigurations method to complete. 
+*      Hence the deadlock.
+*
+* The solution is to remove the WaitForAllAsync from RemoveConfigurations. 
+* This test will deadlock if the WaitForAllAsync function is not removed.
+* 
+*/
+TEST_F(ConfigAdminTests, testManagedServiceRemoveConfigurationsDeadlock)
+{
+  auto f = GetFramework();
+  auto ctx = f.GetBundleContext();
+
+  //Install and start the bundle containing the cm.testdeadlock configuration object.
+  auto bundles = InstallBundles(ctx, "TestBundleManagedServiceDeadlock");
```

Comment:
is there already a bundle which provides the same behavior (a configuration data only bundle)? I'd like to avoid adding more test bundles is possible.